src/artifacts_poetry_plugin/plugin.py

```diff
@@ -4,23 +4,141 @@
 import poetry.console.application
 from poetry.factory import Factory
 from cleo.commands.command import Command
+import poetry.config.config
+from poetry.publishing.uploader import Uploader
+from poetry.utils.authenticator import Authenticator
+from poetry.core.packages.package import Package
+from cleo.helpers import argument
+from typing import List
+import logging
 
-ARTIFACTS_LIST = "artifacts-list"
+ARTIFACTS_DEPLOY = "artifacts-deploy"
+REPOSITORY_NAME = "repository_name"
+logger = logging.getLogger(__name__)
 
 
-def getPackages(poetry):
+def get_repository_data(config, repo_name, io):
+    """Gets the URL, username, and password from the poetry configuration for the given repo name"""
+    repo_data = {}
+    repo_data["url"] = config.get(f"repositories.{repo_name}.url")
+    if repo_data["url"] is None:
+        raise RuntimeError(f"Repository {repo_name} is not defined")
+    authenticator = Authenticator(config, io)
+    auth = authenticator.get_http_auth(repo_name)
+    if auth:
+        repo_data["username"] = auth.username
```

Comment:
A: Instead of a map, we should just create an object that we can return with the appropriate fields.